internal/literals/split.go

```diff
@@ -55,13 +55,13 @@ func shuffleStmts(stmts ...ast.Stmt) []ast.Stmt {
 	return stmts
 }
 
-// Xor encrypt chunks based on key and position
-func encryptChunks(chunks [][]byte, key int) {
+// Encrypt chunks based on key and position
+func encryptChunks(chunks [][]byte, op token.Token, key int) {
```

Comment:
```go
 func encryptChunks(chunks [][]byte, op token.Token, key byte) { 
```

Reply:
It's not safe. The logic of key generation is based on int. If it is truncated to byte earlier, there may be unexpected effects.

Reply:
I would argue that the key should also be a byte, and you can truncate the `byteOffset` to byte before the Xor (doesn't make a difference).

```go
chunk[i] = evalOperator(op, chunk[i], key^byte(byteOffset))
```